x/tokenfactory/keeper/admins_test.go

```diff
@@ -0,0 +1,265 @@
+			admin:     suite.TestAccs[0].String(),
+			valid:     true,
+		},
+	} {
+		suite.Run(fmt.Sprintf("Case %s", tc.desc), func() {
+			// Test minting to admins own account
+			_, err := suite.msgServer.Mint(sdk.WrapSDKContext(suite.ChainA.GetContext()), types.NewMsgMint(tc.admin, sdk.NewInt64Coin(tc.mintDenom, 10)))
+			if tc.valid {
+				addr0bal += 10
+				suite.Require().NoError(err)
+				suite.Require().Equal(suite.GetNeutronZoneApp(suite.ChainA).BankKeeper.GetBalance(suite.ChainA.GetContext(), suite.TestAccs[0], suite.defaultDenom).Amount.Int64(), addr0bal, suite.GetNeutronZoneApp(suite.ChainA).BankKeeper.GetBalance(suite.ChainA.GetContext(), suite.TestAccs[0], suite.defaultDenom))
+			} else {
+				suite.Require().Error(err)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestBurnDenom() {
+	addr0bal := int64(0)
+	suite.Setup()
+
+	// Create a denom.
+	suite.CreateDefaultDenom()
+
+	// mint 10 default token for testAcc[0]
+	suite.msgServer.Mint(sdk.WrapSDKContext(suite.ChainA.GetContext()), types.NewMsgMint(suite.TestAccs[0].String(), sdk.NewInt64Coin(suite.defaultDenom, 10)))
+	addr0bal += 10
+
+	for _, tc := range []struct {
+		desc      string
+		amount    int64
+		burnDenom string
+		admin     string
+		valid     bool
+	}{
+		{
+			desc:      "denom does not exist",
+			amount:    10,
+			burnDenom: "factory/osmo1t7egva48prqmzl59x5ngv4zx0dtrwewc9m7z44/evmos",
+			admin:     suite.TestAccs[0].String(),
+			valid:     false,
+		},
+		{
+			desc:      "burn is not by the admin",
+			amount:    10,
+			burnDenom: suite.defaultDenom,
+			admin:     suite.TestAccs[1].String(),
+			valid:     false,
+		},
+		{
+			desc:      "burn amount is bigger than minted amount",
+			amount:    1000,
+			burnDenom: suite.defaultDenom,
+			admin:     suite.TestAccs[1].String(),
+			valid:     false,
+		},
+		{
+			desc:      "success case",
+			amount:    10,
+			burnDenom: suite.defaultDenom,
+			admin:     suite.TestAccs[0].String(),
+			valid:     true,
+		},
+	} {
+		suite.Run(fmt.Sprintf("Case %s", tc.desc), func() {
+			// Test minting to admins own account
+			_, err := suite.msgServer.Burn(sdk.WrapSDKContext(suite.ChainA.GetContext()), types.NewMsgBurn(tc.admin, sdk.NewInt64Coin(tc.burnDenom, 10)))
+			if tc.valid {
+				addr0bal -= 10
+				suite.Require().NoError(err)
+				suite.Require().True(suite.GetNeutronZoneApp(suite.ChainA).BankKeeper.GetBalance(suite.ChainA.GetContext(), suite.TestAccs[0], suite.defaultDenom).Amount.Int64() == addr0bal, suite.GetNeutronZoneApp(suite.ChainA).BankKeeper.GetBalance(suite.ChainA.GetContext(), suite.TestAccs[0], suite.defaultDenom))
+			} else {
+				suite.Require().Error(err)
+				suite.Require().True(suite.GetNeutronZoneApp(suite.ChainA).BankKeeper.GetBalance(suite.ChainA.GetContext(), suite.TestAccs[0], suite.defaultDenom).Amount.Int64() == addr0bal, suite.GetNeutronZoneApp(suite.ChainA).BankKeeper.GetBalance(suite.ChainA.GetContext(), suite.TestAccs[0], suite.defaultDenom))
```

Comment:
Honestly I didn't even feel like reading these lines, and I guess noone ever would :skull: